Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -444,14 +447,57 @@ open class ChatMessageListVC: _ViewController,
     }
 
     open func chatMessageActionsVCDidFinish(_ vc: ChatMessageActionsVC) {
+        messageActionsVC = nil
         dismiss(animated: true)
     }
 
     // MARK: - ChatMessageContentViewDelegate
 
     open func messageContentViewDidTapOnErrorIndicator(_ indexPath: IndexPath?) {
-        guard let indexPath = indexPath else { return log.error("IndexPath is not available") }
-        didSelectMessageCell(at: indexPath)
+        guard let indexPath = indexPath else {
+            return log.error("IndexPath is not available")
+        }
+        
+        guard let message = dataSource?.chatMessageListVC(self, messageAt: indexPath) else {
+            return log.error("DataSource not found for the message list.")
+        }
+        
+        guard message.isBounced else {
+            didSelectMessageCell(at: indexPath)
+            return
+        }
+        
+        guard let cid = message.cid else {
+            return log.error("Message cid not found.")
+        }
+        
+        let messageController = client.messageController(
+            cid: cid,
+            messageId: message.id
+        )
+
+        messageActionsVC = components.messageActionsVC.init()
+        messageActionsVC?.messageController = messageController
+        messageActionsVC?.channelConfig = dataSource?.channel(for: self)?.config
+        messageActionsVC?.delegate = self
+        
+        guard let messageActions = messageActionsVC?.messageActionsForAlertMenu else {
+            return log.error("messageActionsVC: messageActionsForAlertMenu not found.")
+        }
+        
+        let alert = UIAlertController(title: L10n.Message.Moderation.title, message: L10n.Message.Moderation.message, preferredStyle: .alert)
+        
+        messageActions.forEach { messageAction in
+            alert.addAction(UIAlertAction(title: messageAction.title, style: messageAction.isDestructive ? .destructive : .default) { _ in
+                messageAction.action(messageAction)
+            })
+        }
+        
+        alert.addAction(UIAlertAction(title: L10n.Alert.Actions.cancel, style: .destructive) { _ in
+            self.messageActionsVC = nil
+        })
+                
+        navigationController?.present(alert, animated: true)
```

Comment:
Overall the PR looks good 👍 The only part I didn't quite understand, was this logic. Why is this not handled by the `ChatMessageActionsVC`? At least, as a first impression, I think this logic should be inside the `didSelectMessageCell()`. Can you clarify why it is not? Thanks!